api_scrape.py

```diff
@@ -1,17 +1,31 @@
-from bs4 import BeautifulSoup
-import urllib.request
-
+import requests
+from requests.exceptions import HTTPError
 
 urls = open('appIDs.txt', 'r')
 
-with urls as f:
-    url = f.readline()
-    req = urllib.request.Request(url)
-    resp = urllib.request.urlopen(req)
-    respData = resp.read()
-    print(respData)
 
+try:
+    with urls as f:
+        url = f.readline()
+        response = requests.get(url)
+        response.raise_for_status()
+        json_response = response.json()
+        print(json_response)
+except HTTPError as http_err:
+    print(f'An HTTP error has occurred: {http_err}')
+except Exception as err:
+    print(f'Another error has occurred: {err}')
+
+
+for key, value in json_response.items():
+    categories = json_response[key]['data']['categories']
```

Comment:
Here you use `json_response[key]` when you might not need to. You can probably use `value` which you set in the `for` loop. I believe the `value` holds the actual value of the dictionary item in the current loop iteration